### PR TITLE
Track used Android pointer IDs and reuse IDs after a pointer is released

### LINKED ISSUE #61421: When I use PlatformView with GestureDetector and swipe it quickly with multiple fingers in flutter app, the program crashes.
[The complete Link ](http://note.youdao.com/s/3xniXDek)
[The demo project](https://github.com/liangchaojie/flutterCrashDemo)
[The repeat video] https://github.com/liangchaojie/flutterCrashDemo/blob/master/crash.mp4

### PR DESCRIPTION
**Pull request by jason-simmons**
This avoids increasing pointer ID values beyond the limit accepted
by the Android APIs.

Fixes https://github.com/flutter/flutter/issues/61421

### CODE REVIEW

**Comment by iskakaushik** [comment]
Is it possible to write a test for this?

**Comment by PingShen** [comment]
perfect!!!!
thank you for helping us to avoid this crash

**Comment by cwlp8024** [comment]
Hi @jason-simmons 
Thank you for reading my issues [/61421](https://github.com/flutter/flutter/issues/61421)
I have tried this pull request following your steps and the crash cant reproduce.
Thank you very much again!

File: packages/flutter/lib/src/services/platform_views.dart [hunk 1]
```diff
@@ -503,14 +504,18 @@ class _AndroidMotionEventConverter {
...
   int downTimeMillis;
 
   void handlePointerDownEvent(PointerDownEvent event) {
-    if (nextPointerId == 0) {
+    if (pointerProperties.isEmpty) {
       downTimeMillis = event.timeStamp.inMilliseconds;
     }
-    pointerProperties[event.pointer] = propertiesFor(event, nextPointerId++);
+    int androidPointerId = 0;
+    while (usedAndroidPointerIds.contains(androidPointerId)) {
```
Comment (blasten) [inline packages/flutter/lib/src/services/platform_views.dart:514]: qq: how does this ensure that pointer id doesn't reach the limit? As pointed out in https://github.com/flutter/flutter/issues/61421#issuecomment-659715358

Comment (jason-simmons) [reply packages/flutter/lib/src/services/platform_views.dart:514]: The `MotionEvent` pointer ID limit is an internal Android implementation detail that AFAICT isn't documented anywhere.

This patch ensures that the pointer ID value will be less than the number of active pointers.  The version of Android that I looked at apparently accepts pointer IDs in the range 0 through 31.  So realistically this patch will keep the pointer ID values within a safe range given that a user is unlikely to have >32 pointers active simultaneously.

Given that it isn't a defined part of Android's API I didn't want to try to hardcode a limit here.  But we could enforce a maximum number of active pointers on the framework side if this is a concern.

**Review by iskakaushik — APPROVED** [review]
LGTM modulo @blasten 's open question.

**Pull request merged**